plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -92,20 +94,33 @@ export class EventsProcessor {
                         this.pluginsServer.statsd?.timing('kafka_queue.single_save.snapshot', singleSaveTimer, {
                             team_id: teamId.toString(),
                         })
-                        // No return value in case of snapshot events as we don't do action matching on them
                     } finally {
                         clearTimeout(timeout2)
                     }
                 }
+            } else if (data['event'] === '$performance_event') {
+                const timeout3 = timeoutGuard('Still running "createPerformanceEvent". Timeout warning after 30 sec!', {
+                    eventUuid,
+                })
+                try {
+                    await this.createPerformanceEvent(eventUuid, teamId, distinctId, timestamp, ip, properties)
+                    // No return value in case of performance events as we don't do further processing on them
+
+                    this.pluginsServer.statsd?.timing('kafka_queue.single_save.performance', singleSaveTimer, {
+                        team_id: teamId.toString(),
+                    })
+                } finally {
+                    clearTimeout(timeout3)
+                }
```

Comment:
```suggestion
                const performanceEventTimeout = timeoutGuard('Still running "createPerformanceEvent". Timeout warning after 30 sec!', {
                    eventUuid,
                })
                try {
                    await this.createPerformanceEvent(eventUuid, teamId, distinctId, timestamp, ip, properties)
                    // No return value in case of performance events as we don't do further processing on them

                    this.pluginsServer.statsd?.timing('kafka_queue.single_save.performance', singleSaveTimer, {
                        team_id: teamId.toString(),
                    })
                } finally {
                    clearTimeout(performanceEventTimeout)
                }
```